server/controllers/entities/lib/resolver/resolve_edition.ts

```diff
@@ -1,33 +1,33 @@
 import { getEntitiesByIsbns } from '#controllers/entities/lib/get_entities_by_isbns'
+import type { SanitizedResolverEntry } from '#server/types/resolver'
 import { resolveExternalIds } from './resolve_external_ids.js'
 // Do not try to resolve edition on Wikidata while Wikidata editions are in quarantine
 // cf https://github.com/inventaire/inventaire/issues/182
 const resolveOnWikidata = false
 
-export default entry => {
+export async function resolveEdition (entry: SanitizedResolverEntry) {
   const { isbn, claims } = entry.edition
 
-  return Promise.all([
+  const [
+    uriFoundByIsbn,
+    resultsFoundByExternalIds,
+  ] = await Promise.all([
     resolveByIsbn(isbn),
-    resolveExternalIds(claims, resolveOnWikidata),
+    resolveExternalIds(claims, { resolveOnWikidata }),
   ])
-  .then(pickUriFromResolversResponses)
-  .then(uri => {
-    if (uri != null) { entry.edition.uri = uri }
-    return entry
-  })
+  // TODO: handle possible conflict between uriFoundByIsbn and urisFoundByExternalIds
+  let uri
+  if (uriFoundByIsbn) {
+    uri = uriFoundByIsbn
+  } else if (resultsFoundByExternalIds && resultsFoundByExternalIds.length === 1) {
```

Comment:
```suggestion
  } else if (isNonEmptyArray(resultsFoundByExternalIds)) {
```

[non-blocking]

Reply:
I think that `if (resultsFoundByExternalIds.length > 1)` it should not resolve, as we would not which of the found editions is the right one